guava/src/com/google/common/base/Converter.java

```diff
@@ -165,12 +167,23 @@ protected Converter() {
    * @return the converted value; is null <i>if and only if</i> {@code a} is null
    */
   @CanIgnoreReturnValue
-  public final @Nullable B convert(@Nullable A a) {
+  public final @PolyNull B convert(@PolyNull A a) {
     return correctedDoForward(a);
   }
 
-  @Nullable
-  B correctedDoForward(@Nullable A a) {
+  @SuppressWarnings({
+    "return.type.incompatible",
+    /*
```

Comment:
Your code is inconsistent about whether it uses /* or // style comments.  I suggest // except for Javadoc.  It's easier to read and more compact.